web-common/src/features/canvas/components/time-filter-store.ts

```diff
@@ -0,0 +1,62 @@
+import {
+  useAllDimensionFromMetric,
+  useAllMeasuresFromMetric,
+} from "@rilldata/web-common/features/canvas/components/selectors";
+import type { StateManagers } from "@rilldata/web-common/features/canvas/state-managers/state-managers";
+import { getValidFilterForMetricView } from "@rilldata/web-common/features/dashboards/stores/filter-utils";
+import type { DashboardTimeControls } from "@rilldata/web-common/lib/time/types";
+import type { V1TimeRange } from "@rilldata/web-common/runtime-client";
+import { derived, type Writable } from "svelte/store";
+
+// This helper returns a derived store that yields the final timeRange and where clause
+export function createTimeAndFilterStore(
+  ctx: StateManagers,
+  instanceId: string,
+  metricsViewName: string,
+  {
+    timeRangeStore,
+    overrideTimeRange,
```

Comment:
Could you add a comment to explain when the `overrideTimeRange` comes into play?